pkg/controller/tokenrequestor/reconciler.go

```diff
@@ -252,11 +256,16 @@ func tokenExpirationSeconds(secret *corev1.Secret) (int64, error) {
 	return int64(expirationDuration / time.Second), nil
 }
 
-func getServiceAccountFromAnnotations(annotations map[string]string) *corev1.ServiceAccount {
+func (r *Reconciler) getServiceAccountFromAnnotations(annotations map[string]string) *corev1.ServiceAccount {
+	namespace := r.TargetNamespace
+	if namespace == "" {
+		namespace = annotations[resourcesv1alpha1.ServiceAccountNamespace]
+	}
```

Comment:
Does it make sense to add another test case for this scenario?

Reply:
Done